docs/build/html/_sources/threads.rst

```diff
@@ -0,0 +1,44 @@
+Threading
+---------
+
+We have implemented multithreading into the project for effeciency.
+The way we have done this is by creating classes of the two models used
+in this project, so they many return an instance of a thread to be used in the
+``image_processer.py`` file.
+
+    The Cohere LLM does not have a multihreading implementation, as it needs to wait for the input
+    from both the easyOCR and blip-image models.
+
+- ``blip_image_thread.py``
+    Class that represents a thread instance salesforce BLIP image captioning model. Includes
```

Comment:
The writing you've done overall is well done. I would recommend taking the docstrings from each method in these classes and expanding on them here. That is, take the docstrings that explains what each method is and expand on them as you see fit. If the docstrings are written well enough, they can be put here as is.